app/javascript/forms/provider-forms/cloud-provider-form/index.jsx

```diff
@@ -0,0 +1,170 @@
+  {
+    component: componentTypes.SELECT,
+    name: 'zone_name',
+    label: __('Zone'),
+    placeholder: `<${__('Choose')}>`,
+    loadOptions: loadProviderServerZones,
+    isRequired: true,
+    validate: [{
+      type: validatorTypes.REQUIRED,
+    }],
+  },
+];
+
+const CloudProviderForm = ({ providerId, kind, redirect }) => {
+  const [{ type, schema, values }, setState] = useState({ schema: { fields: [] } });
+
+  const loadProviderSchema = (type, newValues = {}) => {
+    API.options(`/api/providers?type=${type}`).then(({ data: { provider_form_schema } }) => { // eslint-disable-line camelcase
+      setState({
+        type,
+        schema: {
+          fields: [
+            ...initialSchema,
+            {
+              component: componentTypes.SUB_FORM,
+              name: type,
+              ...provider_form_schema, // eslint-disable-line camelcase
+            },
+          ],
+        },
+        values: { type, ...newValues },
+      });
+    }).then(miqSparkleOff);
+  };
+
+  useEffect(() => {
+    if (providerId) {
+      miqSparkleOn();
+      API.get(`/api/providers/${providerId}?attributes=endpoints,authentications,zone_name`).then(({
+        type,
+        endpoints: _endpoints,
+        authentications: _authentications,
+        ...provider
+      }) => {
+        const endpoints = _endpoints.reduce((obj, { role, ...endpoint }) => ({
+          ...obj,
+          [role]: endpoint,
+        }), {});
+
+        const authentications = _authentications.reduce((obj, { authtype, ...authentication }) => ({
+          ...obj,
+          [authtype]: authentication,
+        }), {});
+
+        loadProviderSchema(type, { ...provider, endpoints, authentications });
+      });
+    }
+  }, [providerId]);
+
+  const typeSelected = ({ active, values: { type: newType } = {} }) => {
+    if (active === 'type' && type !== newType) {
+      miqSparkleOn();
+      loadProviderSchema(newType);
+    }
+  };
+
+  const onCancel = () => {
+    const message = sprintf(providerId ? __('Edit of %s was cancelled by the user') : __('Add of %s was cancelled by the user'), kind);
+    miqRedirectBack(message, 'success', redirect);
+  };
+
+  const onSubmit = ({ type, ..._data }, { getState }) => {
+    miqSparkleOn();
+
+    const message = sprintf(__('%s %s was saved'), kind, values.name);
+
+    // Retrieve fields from the schema, but omit the validator components as the API doesn't like them
+    const fields = Object.keys(getState().modified).filter(field => !field.match(/^authentications\.[^.]+\.valid$/));
+    // Filter out fields that are not available in the form schema
+    const data = { ..._.pick(_data, fields), ddf: true };
+
+    const request = providerId ? API.patch(`/api/providers/${providerId}`, data) : API.post('/api/providers', { type, ...data });
+
+    request.then(() => miqRedirectBack(message, 'success', redirect)).catch(miqSparkleOff);
+  };
+
+  return (
+    <div>
+      { !providerId && (
+        <MiqFormRenderer
```

Comment:
I am starting to suspect that the issue with the empty initial values may be caused by these two form instances which have the same field name. Can you try using the select without the renderer? It is exported in `rawComponents` object (i know stupid name..)

Reply:
I don't really understand what you're asking. But IMO having 2 form instances with the same field should be a valid workflow.

Reply:
It is, and the state should be isolated, but can you just try it out?

Reply:
Can I just simply get rid of the first form for a test run?

Reply:
yup